MDANSE/Src/MDANSE/Mathematics/Signal.py

```diff
@@ -292,3 +300,757 @@ def get_spectrum(signal, window=None, timeStep=1.0, axis=0, fft="fft"):
+        """Set up the default filter settings."""
+        cls.default_settings = {
+            "fundamental_freq": {
+                "description": "Spacing between filter peaks (value must evenly divide sample frequency)",
+                "value": DEFAULT_FILTER_CUTOFF,
+            },
+            "quality_factor": {
+                "description": "Specifies bandwidth, proportional to time taken for filter to decay by a factor of 1/e",
+                "value": 1.0,
+            },
+            "comb_type": {
+                "description": "Determines whether quality factor applies to notches or peaks",
+                "values": {"peak", "notch"},
+                "value": "notch",
+            },
+            "pass_zero": {
+                "description": "Determines whether notches or peaks centered on integer multiples of fundamental frequency",
+                "values": {True, False},
+                "value": False,
+            },
+        }
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        self.coeffs = self.TransferFunction(
+            *signal.iircomb(
+                self.fundamental_freq,
+                self.quality_factor,
+                ftype=self.comb_type,
+                pass_zero=self.pass_zero,
+                fs=self.sample_freq,
+            )
+        )
+        self.freq_response = (self.coeffs, self.__class__.FrequencyRangeMethod.FFT)
+
+    def compute_frequencies(
+        self, transfer_function: Filter.TransferFunction, range: np.ndarray
+    ):
+        """ """
+        return signal.freqz(*transfer_function, worN=range, fs=self.sample_freq)
+
+
+FILTERS = (
+    Butterworth,
+    ChebyshevTypeI,
+    ChebyshevTypeII,
+    Elliptical,
+    Bessel,
+    Notch,
+    Peak,
+    Comb,
+)
+
+filter_map = {filter_class.__name__: filter_class for filter_class in FILTERS}
+
+
+def power_spectrum(
+    trajectory, frames, projection, atom_selection, weights, instrument_resolution
+) -> Tuple[np.ndarray, np.ndarray]:
+    """Returns the position power spectrum of a configuration's constituent atomic trajectories.
+
+    :Parameters:
+        #. trajectory ( ): atomic trajectory object
+        #. frames ( ):
+        #. projection ( ):
+        #. atom_selection ( ):
+        #. weights ( ):
+        #. instrument_resolution ( ):
+    :Returns:
+        #. Tuple[np.ndarray, np.ndarray]: tuple containing the omegas (frequency range) and the corresponding power spectrum of the atomic trajectories
+    """
+    num_atoms = len(atom_selection["indices"])
+    trajectory = trajectory["instance"]
+    sorted_atoms = trajectory.chemical_system.atom_list
+
+    output = dict()
+    output["romega"] = instrument_resolution["romega"]
+
+    for element in atom_selection["unique_names"]:
+        output["pacf_%s" % element] = np.zeros(
+            np.array(range(frames["n_frames"])).shape
+        )
+        output["pps_%s" % element] = np.zeros(output["romega"].shape)
```

Comment:
```suggestion
    for element in atom_selection["unique_names"]:
        output[f"pacf_{element}"] = np.zeros(frames["n_frames"])
        output[f"pps_{element}"] = np.zeros_like(output["romega"])
```
What is the logic behind:
```np.array(range(frames["n_frames"])).shape```
That's just a 1D array of length `frames["n_frames"]`